src/utils/vvim.cpp

```diff
@@ -2047,12 +2125,14 @@ bool VVim::handleKeyPressEvent(int key, int modifiers, int *p_autoIndentPos)
                 addRangeToken(range);
                 processCommand(m_tokens);
                 break;
-            } else if (hasActionToken() || !checkMode(VimMode::Normal)) {
-                // Invalid sequence.
-                break;
-            } else if (checkPendingKey(Key(Qt::Key_BracketLeft))) {
+            } else if (hasActionToken() && checkPendingKey(Key(Qt::Key_BracketLeft))) {
```

Comment:
`hasActionToken()`? `[{` now does not go to the upper level title. It just trigger the wrong action `ParagraphUp` instead. :(

Reply:
I see, I think it's hard to follow the vim spec and in same time add our own spec. So decided to close this PR, and make it as my own customization. Thanks for your help.

Reply:
@xianzhon :) I have implemented these 3 operations according to yours (9ef59990dc6eee79d0b219bb625d51ab4df055ce). Thanks very much!

Reply:
@tamlok Nice! I read your commmit, your implementation is more like the vim style (I'm now reading the vim source code) and must be more efficient too! Good!!